src/renderer/port-forward/port-forward.store.ts

```diff
@@ -0,0 +1,218 @@
+import { waitUntilFree } from "tcp-port-used";
+import { Notifications } from "../components/notifications";
+import logger from "../../common/logger";
+
+export interface ForwardedPort {
+  clusterId?: string;
+  kind: string;
+  namespace: string;
+  name: string;
+  port: string;
+  forwardPort: string;
+}
+
+export class PortForwardStore extends ItemStore<PortForwardItem> {
+  private storage = createStorage<ForwardedPort[] | undefined>("port_forwards", undefined);
+
+  constructor() {
+    super();
+    makeObservable(this);
+    autoBind(this);
+
+    this.init();
+  }
+
+  private async init() {
+    await this.storage.whenReady;
+
+    const savedPortForwards = this.storage.get(); // undefined on first load
+
+    if (Array.isArray(savedPortForwards)) {
+      logger.info("[PORT_FORWARD] starting saved port-forwards");
+      await Promise.all(savedPortForwards.map(pf => {
+        const port = new PortForwardItem;
+
+        port.clusterId = pf.clusterId;
+        port.kind = pf.kind;
+        port.namespace = pf.namespace;
+        port.name = pf.name;
+        port.port = pf.port;
+        port.forwardPort = pf.forwardPort;
+
+        return addPortForward(port);
+      }));
+
+      this.portForwards = [];
+    }
+  }
+
+  @observable selectedItemId?: string;
+  @observable portForwards: PortForwardItem[];
+
+  @computed get selectedItem() {
+    return this.portForwards.find((e: ItemObject) => e.getId() === this.selectedItemId);
+  }
+
+  watch() {
+    const disposers: IReactionDisposer[] = [
+      reaction(() => this.portForwards, () => this.loadAll()),
+    ];
+
+    return () => disposers.forEach((dispose) => dispose());
+  }
+
+  loadAll() {
+    return this.loadItems(async () => {
+      let portForwards = await getPortForwards();
+
+      // filter out any not for this cluster
+      portForwards = portForwards?.filter(pf => pf.clusterId == getHostedClusterId());
+      this.storage.set(portForwards);
+
+      this.reset();
+      portForwards?.forEach(pf => {
+        const port = new PortForwardItem;
+
+        port.clusterId = pf.clusterId;
+        port.kind = pf.kind;
+        port.namespace = pf.namespace;
+        port.name = pf.name;
+        port.port = pf.port;
+        port.forwardPort = pf.forwardPort;
+        this.portForwards.push(port);
+      });
+
+      return this.portForwards;
+    });
+  }
+
+  reset() {
+    this.portForwards = [];
+  }
+
+  async removeSelectedItems() {
+    return Promise.all(this.selectedItems.map(removePortForward));
+  }
+}
+
+interface PortForwardResult {
+  port: number;
+}
+
+interface PortForwardsResult {
+  portForwards: ForwardedPort[];
+}
+
+export async function addPortForward(portForward: ForwardedPort): Promise<number> {
+  let response: PortForwardResult;
+
+  try {
+    response = await apiBase.post<PortForwardResult>(`/pods/${portForward.namespace}/${portForward.kind}/${portForward.name}/port-forward/${portForward.port}/${portForward.forwardPort}`, {});
```

Comment:
This seems error prone, wouldn't it be better to use the `buildURL` methods directly?